### ISSUE TITLE
Implement tp_repr and tp_str

### ISSUE THREAD

**Issue by youknowone** [RustPython/RustPython#4653]
part of #3244 

The slot field to start: https://github.com/RustPython/RustPython/blob/7744e95aea79179f3dc31f9c3ab46e22fd3b5cbf/vm/src/types/slot.rs#L47

In Python side, `str(1)` hits this line :
https://github.com/RustPython/RustPython/blob/7744e95aea79179f3dc31f9c3ab46e22fd3b5cbf/vm/src/builtins/str.rs#L281

`input.str(vm)` in the line is calling rust side rust side `PyObjectRef::str` which is:
https://github.com/RustPython/RustPython/blob/7744e95aea79179f3dc31f9c3ab46e22fd3b5cbf/vm/src/protocol/object.rs#L331-L349

And `object.__str__`, which every other object's `__str__` inherits as default is:
https://github.com/RustPython/RustPython/blob/7744e95aea79179f3dc31f9c3ab46e22fd3b5cbf/vm/src/builtins/object.rs#L184-L188

Once this is done, a wrapper need to be added to `update_slot!` to connect python side defined magic methods to rust type slots.
https://github.com/RustPython/RustPython/blob/7744e95aea79179f3dc31f9c3ab46e22fd3b5cbf/vm/src/types/slot.rs#L349

Once those are finished, we can actually replace the implementation of other types' `__str__` to this new implementation
We typically did it using trait. 
One of the good example is Hashable: https://github.com/RustPython/RustPython/blob/7744e95aea79179f3dc31f9c3ab46e22fd3b5cbf/vm/src/types/slot.rs#L645-L664

And the adaption is looking like this: 
https://github.com/RustPython/RustPython/blob/7744e95aea79179f3dc31f9c3ab46e22fd3b5cbf/vm/src/builtins/complex.rs#L439-L453

**Comment by haxelion** [comment]
@youknowone Please assign me 🙂

**Comment by arihant2math** [comment]
Completed, and superseded by #3244.
